### PR TITLE
INT 1746 Point Intake to Ferb Screening Create

### PR DESCRIPTION
**Pull request by TylerOSI**
### Jira Story

- [Name of Story INT-1746](https://osi-cwds.atlassian.net/browse/INT-1746)

## Description
In the effort to phase out Intake API we need to create screenings using the new Ferb endpoint.

Originally, we were going to switch create, update, and get to Ferb at the same time. Since there are some complexities due to changes in api implementation and contract, we are planning to roll them out one at a time. That will make for smaller stories and smaller PRs.
## Tests
- [x] I have included unit tests 
- [ ] I have included feature tests 
- [ ] I have included other tests 
- [ ] I have NOT included tests 

## Types of changes

- [ ] New feature (non-breaking change which adds functionality)
- [x] Refactoring (No behavioral changes)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] I have ran all tests for the project.
- [x] I have ran lint/rubocop check for the changed/new files.
- [x] My code is in a stable state ready to be deployable, but not necessarily complete.
- [x] I promise on my honor as a CWDS developer to ensure I don't break the build, to check Lint/Rubocop, use best practices, to leave the code in better shape than I found it, and to have a good day.

### CODE REVIEW

File: app/repositories/screening_repository.rb [hunk 1]
```diff
@@ -4,13 +4,13 @@
...
 class ScreeningRepository
   def self.create(security_token, screening)
-    response = IntakeAPI.make_api_call(
+    response = FerbAPI.make_api_call(
       security_token,
-      ExternalRoutes.intake_api_screenings_path,
+      FerbRoutes.intake_screenings_path,
       :post,
-      screening.as_json.except('id')
+      screening.as_json(except: 'id')
```
Comment (wmitchell-cambria) [inline app/repositories/screening_repository.rb]: What's the difference between these?

File: spec/controllers/api/v1/screenings_controller_spec.rb [hunk 2]
```diff
@@ -19,194 +19,262 @@
...
+          cross_reports: [],
+          participants: [],
+          allegations: []
+        }
+      end
 
       it 'is blank if user_details is empty' do
         staff = FactoryBot.build(:staff, first_name: nil, last_name: nil, county: nil)
         session = { 'security_token' => security_token, 'user_details' => staff }
-        expect(Screening).to receive(:new)
```
Comment (wmitchell-cambria) [inline spec/controllers/api/v1/screenings_controller_spec.rb]: Do we still assert anywhere in this test that the response is blank?

File: spec/controllers/api/v1/screenings_controller_spec.rb [hunk 3]
```diff
@@ -19,194 +19,262 @@
...
-            indexable: true
-          )
-          .and_return(blank_screening)
         process :create, method: :post, session: session
         expect(response).to be_successful
       end
+    end
+
+    describe 'setting assignee when user_details is set' do
+      let(:created_screening) do
```
Comment (wmitchell-cambria) [inline spec/controllers/api/v1/screenings_controller_spec.rb]: This is a little nitpicky, but it seems odd to have this `:created_screening` defined here, and then have several subsuites redefine it. It's a little hard to follow, and could be a sign that these tests should be regrouped. Oftentimes, we'll end up with more concise test files if we group by setup rather than behavior. (I can expound upon that in some more suitable format at a later date.)

File: spec/controllers/api/v1/screenings_controller_spec.rb [hunk 4]
```diff
@@ -19,194 +19,262 @@
...
+        {
+          reference: '123ABC',
+          assignee: "#{staff.first_name} Q. #{staff.last_name} - #{staff.county}",
+          assignee_staff_id: staff.staff_id.to_s,
+          incident_county: nil,
+          indexable: true,
```
Comment (wmitchell-cambria) [inline spec/controllers/api/v1/screenings_controller_spec.rb]: And I guess while I'm commenting on that in general, is there a way to combine some of the duplicated empty fields of all of these? Having large hashes where most things are the same makes it harder to follow what is changing from test to test.

File: spec/controllers/api/v1/screenings_controller_spec.rb [hunk 5]
```diff
@@ -19,194 +19,262 @@
...
+            cross_reports: [],
+            participants: [],
+            allegations: []
+          }
+        end
 
-      describe 'when user_details is set' do
         it 'formats assignee as first mi. last - county if all exist' do
-          staff = FactoryBot.build(:staff, middle_initial: 'Q', staff_id: '456')
-          assignee = "#{staff.first_name} Q. #{staff.last_name} - #{staff.county}"
```
Comment (wmitchell-cambria) [inline spec/controllers/api/v1/screenings_controller_spec.rb]: I'm not following where we are actually testing that the assignee is formatted. It looks like this test now just asserts that it gets some 2xx response.

File: spec/controllers/api/v1/screenings_controller_spec.rb [hunk 6]
```diff
@@ -19,194 +19,262 @@
...
+            addresses: [],
+            cross_reports: [],
+            participants: [],
+            allegations: []
+          }
+        end
 
         it 'formats assignee as first last - county if no middle initial' do
-          staff = FactoryBot.build(:staff, staff_id: '789')
-          assignee = "#{staff.first_name} #{staff.last_name} - #{staff.county}"
```
Comment (wmitchell-cambria) [inline spec/controllers/api/v1/screenings_controller_spec.rb]: Same here. Do we actually test the behavior asserted in the `it` statement?

File: app/controllers/api/v1/screenings_controller.rb [hunk 7]
```diff
@@ -43,13 +43,6 @@ class ScreeningsController < ApiController # :nodoc:
       ].freeze
 
       def create
-        new_screening = Screening.new(
```
Comment (wmitchell-cambria) [inline app/controllers/api/v1/screenings_controller.rb]: Can you discuss some of your reasoning behind no longer using the Screening model?

File: spec/controllers/api/v1/screenings_controller_spec.rb [hunk 8]
```diff
@@ -19,194 +19,262 @@
...
+        indexable: true,
+        incident_address: {},
+        addresses: [],
+        cross_reports: [],
+        participants: [],
+        allegations: []
+      }
+    end
+
+    let(:blank_screening) do
```
Comment (wmitchell-cambria) [inline spec/controllers/api/v1/screenings_controller_spec.rb]: Is this used anywhere?

File: spec/controllers/api/v1/screenings_controller_spec.rb [hunk 9]
```diff
@@ -19,194 +19,262 @@
...
+        cross_reports: [],
+        participants: [],
+        allegations: []
+      }
+    end
     before do
       allow(LUID).to receive(:generate).and_return(['123ABC'])
       expect(ScreeningRepository).to receive(:create)
-        .with(security_token, blank_screening)
+        .with(security_token, created_screening)
```
Comment (wmitchell-cambria) [inline spec/controllers/api/v1/screenings_controller_spec.rb]: I just want to confirm here that Ruby works differently than JavaScript in how it binds `created_screening` here. Each `it` block will use the most specific available `created_screening` to itself, rather than the one defined a few lines up from here, correct?

File: app/repositories/screening_repository.rb (hunk 1 above)
Comment (TylerOSI) [reply app/repositories/screening_repository.rb]: there is not a difference functionally -- they do the same thing.

File: spec/controllers/api/v1/screenings_controller_spec.rb (hunk 3 above)
Comment (TylerOSI) [reply spec/controllers/api/v1/screenings_controller_spec.rb]: I see. A few observations on my side. First, I don't need to define `created_screening` here if its repeated in each test case -- so it can be removed. Second, since the behavior is different for each test case, I would prefer to have a separate stubbed response for each one of them and I would like to keep it as close to the test case as possible.

File: spec/controllers/api/v1/screenings_controller_spec.rb (hunk 4 above)
Comment (TylerOSI) [reply spec/controllers/api/v1/screenings_controller_spec.rb]: I could abstract out some of the fields that are staying the same. Though, I think there is some benefit to having simple request / response JSON in one place -- you don't have to look around to find what the request / response is. Also, there isn't a very logical grouping for these fields that aren't changing.

File: spec/controllers/api/v1/screenings_controller_spec.rb (hunk 2 above)
Comment (TylerOSI) [reply spec/controllers/api/v1/screenings_controller_spec.rb]: as mentioned in one of my bellow comments, I will work on fixing that too.

File: spec/controllers/api/v1/screenings_controller_spec.rb (hunk 5 above)
Comment (TylerOSI) [reply spec/controllers/api/v1/screenings_controller_spec.rb]: yeah, ill update the expectations.

File: app/controllers/api/v1/screenings_controller.rb (hunk 7 above)
Comment (TylerOSI) [reply app/controllers/api/v1/screenings_controller.rb]: In a Rails application, we typically use models to help with a relational database model. In Intake, there is no relational database model (yes we use postgres, but Ferb is now maintaining that). Ferb, and other APIs, define the contract between intake and the API layer and also perform request validation. All we need to do is send requests to Ferb in JSON format from intake. The screening model, is an extra layer of formatting and potentially rails validation that we don't need.

That said, IMO the models should be kept around for now since the factories use them and we need factories in our feature specs for easily stubbing API calls with built requests.

File: spec/controllers/api/v1/screenings_controller_spec.rb (hunk 9 above)
Comment (TylerOSI) [reply spec/controllers/api/v1/screenings_controller_spec.rb]: yeah that is correct.
